tests/integration/oci_client.py

```diff
@@ -0,0 +1,185 @@
+import hashlib
```

Comment:
The code in this file looks good to me. I am just wondering whether we need this at all. It goes pretty low level, and existing clients already exist (skopeo, podman, oras, cosign). Have you considered any of these? If so, are there not sufficient for this use case?